docs/advanced/typescript.mdx

```diff
@@ -38,8 +38,12 @@ declare module '@mdx-js/react' {
     | 'tr'
     | 'ul'
 
+  export type ComponentProps = React.HTMLProps<Element> & {
+    metastring?: string
+  }
+
   export type Components = {
-    [key in ComponentType]?: React.ComponentType<{children: React.ReactNode}>
+    [key in ComponentType]?: React.ComponentType<ComponentProps>
```

Comment:
I'm not sure this is accurate.
The default components are based of html elements, but it can be extended with additional elements which may not be HTML tag based.

Reply:
For example `inlineCode` and `thematicBreak` are both not directly HTML types, but are part of the `ComponentType` list, which this builds on.

Reply:
Thanks for reviewing this!
Yes, it is not accurate enough. I can replace `ComponentProps` with `any` like [v2](https://github.com/mdx-js/mdx/blob/v2/packages/react/types/index.d.ts) does. Would that be alright?

Reply:
Sure, that could work.

Reply:
cool, done